src/resmom/linux/alps.c

```diff
@@ -725,12 +796,10 @@ stack_busted(ud_t *d)
 			}
 		} else if (strcmp(BASIL_ELM_RSVNARRAY, top) == 0) {
 			if (strcmp(BASIL_ELM_INVENTORY, prev) != 0) {
-				parse_err_illegal_start(d);
-				return (1);
-			}
-			if (brp->data.query.type != basil_query_inventory) {
-				parse_err_illegal_start(d);
-				return (1);
+				if (strcmp(BASIL_ELM_RESPONSEDATA, prev) != 0) {
+					parse_err_illegal_start(d);
+					return (1);
+				}
```

Comment:
I would need some explanation to understand why we are checking for response data before bailing out.